spec/inputs/tcp_spec.rb

```diff
@@ -646,6 +650,75 @@ def get_port
               expect(result.first.get("message")).to eq(message)
             end
           end
+
+          context "with enforced protocol version" do
+            let(:config) do
+              base_config.merge 'ssl_supported_protocols' => [ tls_version ]
+            end
+
+            # when (Docker) testing against LS 6.x it's using Java 1.8 where TLS 1.3 isn't available
+            let(:tls_version) { (JOpenSSL::VERSION > '0.12' && TcpHelpers.tls13_available_by_default?) ? 'TLSv1.3' : 'TLSv1.2' }
+
+            it "should be able to connect and write data" do
+              used_tls_protocol = nil
+              result = TcpHelpers.pipelineless_input(subject, 1) do
+                sslsocket.connect
+                sslsocket.write("#{message}\n")
+                used_tls_protocol = sslsocket.session.to_java(javax.net.ssl.SSLSession).getProtocol
+                tcp.flush
+                sslsocket.close
+                tcp.close
+              end
+              expect(result.size).to eq(1)
+              expect(used_tls_protocol).to eql tls_version
+            end
+          end
+
+          context "with enforced protocol range" do
+            let(:config) do
+              base_config.merge 'ssl_supported_protocols' => [ 'TLSv1.3', 'TLSv1.2' ]
+            end
+            let(:sslcontext) do
+              super().tap { |ctx| ctx.ssl_version = 'TLSv1.2' }
```

Comment:
comment for clarity: I believe this `sslcontext` is our _external_ SSL that will be establishing a connection to our input plugin, and that we are setting its ssl_version to something that is _included_ in the range supported by our input?

If so, I would hope for another spec, where our external SSL specifies a version that cannot be satisfied by our input, even though that would be largely spec-ing the error as apparent to the external, because we want to make sure that a mismatch doesn't crash our plugin.

Reply:
that's a bit of a redundant spec (pretty much testing the underlying SSL layer) but added as requested ...